lib/std/time/time.go

```diff
@@ -65,6 +65,21 @@ func DoWithClock(shim Clock, fn func() error) error {
 	return fn()
 }
 
+type CleanUpRegisterable interface {
+	Cleanup(func())
+}
+
+// DoWithClockForTest temporarily sets the shim as the time and runs the given function for a test and when the test
+// is done, the time is returned to its original state.
+func DoWithClockForTest(shim Clock, t CleanUpRegisterable) {
```

Comment:
`Do` prefix usually hints at taking a closure.  I'd probably call it `ShimClockForTestingT` or similar.  (Quite a few "ForTestingT" functions around already.)

I'd probably put the testing.T as first arg.

Reply:
Changed the name and made the cleanup function the first argument, I didn't want to make the argument testing.T since all we need is something that implements the Cleanup function (no need to add testing.T here).